transitions/index.js

```diff
@@ -103,6 +117,83 @@ const processChange = (change, callback) => {
     });
 };
 
+const deleteInfoDoc = (change, callback) => {
+  db.medic.get(`${change.id}-info`, (err, doc) => {
+    if (err) {
+      if (err.statusCode === 404) {
+        // don't worry about deleting a non-existant doc
+        return callback();
+      }
+      return callback(err);
+    }
+    doc._deleted = true;
+    db.medic.insert(doc, callback);
+  });
+};
+
+const getAdminNames = callback => {
+  const options = { key: [ 'user-settings' ], include_docs: true };
+  db.medic.view('medic-client', 'doc_by_type', options, (err, results) => {
+    if (err) {
+      return callback(err);
+    }
+    const admins = results.rows
+      .filter(row => {
+        return row.doc.roles &&
+               (row.doc.roles.includes('_admin') ||
+               row.doc.roles.includes('national_admin'));
+      })
+      .map(row => {
+        const parts = row.id.split(':');
+        return parts.length > 1 && parts[1];
+      })
+      .filter(name => !!name);
+    callback(null, admins);
+  });
+};
+
+const deleteReadDocs = (change, callback) => {
+
+  // we don't know if the deleted doc was a report or a message so
+  // attempt to delete both
+  const possibleReadDocIds = [ 'report', 'message' ]
+    .map(type => `read:${type}:${change.id}`);
+
+  getAdminNames((err, admins) => {
+    if (err) {
+      return callback(err);
+    }
+
+    async.each(
+      admins,
+      (admin, callback) => {
+        const metaDb = db.use(`medic-user-${admin}-meta`);
```

Comment:
Just to double-check (because I have no trust) this is stateless right? We're not modifying `db` with this call?

Reply:
I think so. The documentation says:

> creates a scope where you operate inside name.

https://github.com/dscape/nano#nanousename

This implies to me that the returned value is a scoped db.